platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -27,12 +29,45 @@ @implementation NSExpression (MGLAdditions)
         return { std::string([(NSString *)value UTF8String]) };
     } else if ([value isKindOfClass:NSNumber.class]) {
         NSNumber *number = (NSNumber *)value;
-        if((strcmp([number objCType], @encode(int))) == 0) {
-            return { number.intValue };
-        } else if ((strcmp([number objCType], @encode(double))) == 0) {
-            return { number.doubleValue };
-        } else if ((strcmp([number objCType], @encode(char))) == 0) {
-            return { number.boolValue };
+        NSLog(@"NSNumber %@ (%s)", number, number.objCType);
+
+        // 32/64-bit check based on CGFloat storage size
+        // per https://developer.apple.com/library/ios/documentation/General/Conceptual/CocoaTouch64BitGuide/
+        BOOL thirtyTwoBit = sizeof(CGFloat) == sizeof(float);
+        BOOL sixtyFourBit = sizeof(CGFloat) == sizeof(double);
+        NSAssert((thirtyTwoBit || sixtyFourBit) && ! (thirtyTwoBit && sixtyFourBit), @"Fault in 32/64-bit determination");
```

Comment:
This should be a `static_assert()` so the build would fail if this turned out to be false for some reason.